### PR TITLE
style: optimize log output

### PR DESCRIPTION
**Pull request by evanli18**
- replace zap with klog
- set job log level to 4
- ignore NotFound

Closes: #229
Signed-off-by: Evan Li <evan.li97@outlook.com>

### CODE REVIEW

File: controllers/terraform/logging.go [hunk 1]
```diff
@@ -43,6 +43,6 @@ func getPodLog(ctx context.Context, client kubernetes.Interface, namespace, jobN
 		return "", err
 	}
 	logContent := buf.String()
-	klog.Info("pod logs", "Pod", pod.Name, "Logs", logContent)
+	klog.V(4).Info("pod logs", "Pod", pod.Name, "Logs", logContent)
```
Comment (zzxwill) [inline controllers/terraform/logging.go:46]: Why is the changed line better?

Comment (evanli18) [reply controllers/terraform/logging.go:46]: Level 2 is usually used prod, which can effectively avoid too much output of container logs

File: main.go [hunk 2]
```diff
@@ -53,9 +54,12 @@ func main() {
 			"Enabling this will ensure there is only one active controller manager.")
 	flag.DurationVar(&syncPeriod, "informer-re-sync-interval", 10*time.Second,
 		"controller shared informer lister full re-sync period")
+	// embed klog
+	klog.InitFlags(nil)
+	defer klog.Flush()
```
[the remark this answers is not in the document]
Comment (evanli18) [reply main.go]: I deleted this line.

klog can automatically refresh the buffer log in the background, which is sufficient in general. https://github.com/kubernetes/klog/blob/872d9d99fe6ae668f84f7352c8d3900ce42e4774/klog.go#L420

File: controllers/terraform/logging.go (hunk 1 above)
Comment (zzxwill) [reply controllers/terraform/logging.go:46]: Cool. Can you share a link on the klog level. I didn't find any useful references.

Comment (evanli18) [reply controllers/terraform/logging.go:46]: Ref：https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md

Comment (zzxwill) [reply controllers/terraform/logging.go:46]: Great! Very useful link ever!

**Review by zzxwill — APPROVED** [review]
Good job!

**Pull request merged**